github/issues_comments.go

```diff
@@ -118,10 +118,11 @@ func (s *IssuesService) CreateComment(ctx context.Context, owner string, repo st
 }
 
 // EditComment updates an issue comment.
+// A non-nil comment.Body must be provided. Other comment fields should be left nil.
```

Comment:
An additional step we can consider doing is checking if any of the `comment` fields other than `Body` are set (especially `ID`), and return an error with helpful text if so.

GitHub API documents that the only input field is `body`, and it is required. See https://developer.github.com/v3/issues/comments/#edit-a-comment:

![image](https://user-images.githubusercontent.com/1924134/38107559-4c29c808-3360-11e8-9a60-b072fc2e3c56.png)

A breaking API change to replace `IssueComment` parameter with a new `IssueCommentEdit` struct containing only the `Body` field is another option, but that's a breaking API change.